torch/distributed/elastic/rendezvous/dynamic_rendezvous.py

```diff
@@ -140,6 +140,18 @@ def _set_timeouts(self, **timeouts: Optional[timedelta]):
             setattr(self, "_" + name, timeout)
 
 
+@dataclass(repr=False, eq=False, frozen=True)
+class RendezvousSettings:
```

Comment:
Why can't we reuse https://github.com/pytorch/pytorch/blob/master/torch/distributed/elastic/rendezvous/api.py#L124 ?

Reply:
First of all, unlike `RendezvousParameters`, `RendezvousSettings` is immutable/read-only. Beyond that `RendezvousSettings` does not map one-to-one with rendezvous parameters.  For instance the endpoint and the backend name are not part of it. However it has two additional attributes related to keep-alive settings (which are constants for now) and in the future it will hold other attributes that will be derived from a combination of passed parameters and the environment in which the rendezvous runs.